back/src/Guide/Api/Types.hs

```diff
@@ -644,6 +642,42 @@ toCHeading h = CHeading
   , chSlug    = H $ headingSlug h
   }
 
+-- | Custom 'CForest' is the same as 'Forest'.
+-- | It needed to customize JSON instances for TOC to be list of object, not list of lists.
+type CForest a = [CTree a]
+
+-- | Custom 'CTree' is the same as 'Tree'.
+-- | It needed to customize JSON instances for TOC to be list of object, not list of lists.
+data CTree a = CNode
+  { cRootLabel :: a
+  , cSubForest :: CForest a
+  } deriving (Show, Generic)
+
+instance A.ToJSON (CTree CHeading) where
+  toJSON (CNode CHeading{..} cSubForest) = A.object
+    [ "content" A..= chContent
+    , "slug"    A..= chSlug
+    , "forest"  A..= cSubForest
```

Comment:
Please ask @avele or @Gigiman about the name, `"forest"` seems like not the best name.